src/worker.cc

```diff
@@ -769,6 +769,17 @@ void from_json(rapidjson::Document& doc, Options& options) {
 
   // costing defaults to none which is only valid for locate
   auto costing_str = rapidjson::get<std::string>(doc, "/costing", "none");
+  // auto_shortest is deprecated and will be turned into
+  // shortest=true costing option. maybe remove auto_shortest in v4?
+  if (costing_str == "auto_shorter") {
+    costing_str = "auto";
+    auto autoshorter_options = rapidjson::GetValueByPointer(doc, "/costing_options/auto_shorter");
+    if (autoshorter_options) {
+      rapidjson::SetValueByPointer(doc, "/costing_options/auto", *autoshorter_options);
+    }
+    rapidjson::Pointer("/costing_options/auto/shortest").Set(doc, true);
+  }
+
```

Comment:
changes costing to `auto` if `auto_shorter` was specified in the request.. should also make sure that all costing options are transferred appropriately

Reply:
all critical cases are tested BTW